src/syscall/forklock.go

```diff
@@ -0,0 +1,38 @@
+//go:build tinygo && linux && !wasip1 && !wasip2 && !darwin && !baremetal
+
+package syscall
+
+// This is the original ForkLock:
+//
+// var ForkLock sync.RWMutex
+//
+// This requires importing sync, but importing sync causes an import loop:
+//
+// package tinygo.org/x/drivers/examples/net/tcpclient
+//        imports bytes
+//        imports io
+//        imports sync
+//        imports internal/task
+//        imports runtime/interrupt
+//        imports device/arm
+//        imports syscall
+//        imports sync: import cycle not allowed
+//
+// So for now, make our own stubbed-out ForkLock that doesn't use sync..
+
+type forklock struct{}
+
+func (f forklock) RLock()   {}
+func (f forklock) RUnlock() {}
+
+var ForkLock forklock
```

Comment:
Don't. You're basically removing all safety here and breaking the API contract.

Reply:
This again was stub code to get the poc working. Since your PR introduces thread safe mutexes we could replace it with

 ```suggestion
var ForkLock sync.RWMutex
```

I think @scottfeldman did this to circumvent some import cycles, not sure if they still hold, I will investigate this.